### PR TITLE
Add group and app filters for the Okta access list sync.

### PR DESCRIPTION
**Pull request by mdwn**
Group and app filters for the Okta access list sync feature have been added to the plugin settings and to the service settings.

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -5671,6 +5671,14 @@ message PluginOktaSyncSettings {
   // app. When empty the plugin will fall back to the legacy behaviour of syncing
   // users from the entre organization.
   string app_id = 5;
+
+  // GroupFilters are filters for which Okta applications to synchronize as access lists.
+  // These are globs/regexes.
+  repeated string group_filters = 6;
+
+  // AppFilters are filters for which Okta applications to synchronize as access lists.
+  // These are globs/regexes.
```
Comment (kimlisa) [inline api/proto/teleport/legacy/types/types.proto:5680]: should these comments be the same?

Comment (mdwn) [reply api/proto/teleport/legacy/types/types.proto:5680]: Whoops, thank you.

File: lib/config/fileconf.go [hunk 2]
```diff
@@ -2384,10 +2392,26 @@ func (o *OktaSync) Parse() (*servicecfg.OktaSyncSettings, error) {
 		return nil, trace.BadParameter("default owners must be set when access list import is enabled")
 	}
 
+	for _, filter := range o.GroupFilters {
+		_, err := utils.CompileExpression(filter)
+		if err != nil {
+			return nil, trace.Wrap(err, "error parsing group filter")
```
Comment (kimlisa) [inline lib/config/fileconf.go:2398]: for these compile errors, can we return the expression that caused the error in the message too?

**Review by kimlisa — APPROVED** [review]
with minor comment

File: api/proto/teleport/legacy/types/types.proto [hunk 3]
```diff
@@ -5671,6 +5671,14 @@ message PluginOktaSyncSettings {
   // app. When empty the plugin will fall back to the legacy behaviour of syncing
   // users from the entre organization.
   string app_id = 5;
+
+  // GroupFilters are filters for which Okta groups to synchronize as access lists.
+  // These are globs/regexes.
```
Comment (hugoShaka) [inline api/proto/teleport/legacy/types/types.proto:5676]: Can you put a regex example? Across Teleport, we have several regexp-like languages, and it's never clear if it's a re2 regexp, PCRE, or something custom that expands `*` and doesn't need `.*` for example.

For IaC this godoc will be used to generate the reference documentation.

File: api/proto/teleport/legacy/types/types.proto [hunk 4]
```diff
@@ -5671,6 +5671,14 @@ message PluginOktaSyncSettings {
...
+  repeated string group_filters = 6;
+
+  // AppFilters are filters for which Okta applications to synchronize as access lists.
+  // These are globs/regexes.
```
Comment (hugoShaka) [inline api/proto/teleport/legacy/types/types.proto:5680]: idem

**Review by hugoShaka — APPROVED** [review]

**Pull request merged**